samples/matter/lock/README.rst

```diff
@@ -163,13 +166,22 @@ Button 1:
 .. matter_door_lock_sample_button1_end
 
 Button 2:
-    Changes the lock state to the opposite one.
+    For nRF52840dk, nRF5340dk and nRF21540dk:
```

Comment:
```suggestion
    For nRF52840 DK, nRF5340 DK and nRF21540 DK:
```
etc. I think we should either use a full build target name or a human-friendly name.